field/field.cpp

```diff
@@ -16,33 +37,47 @@ Tile Field::operator()(int i, int j) {
     return *_field[i][j];
 }
 
-Field::~Field() {
-    for (int i = 0; i < _field.size(); ++i) {
-        for (int j = 0; j < _field[i].size(); ++j) {
-            delete _field[i][j];
-            _field[i][j] = nullptr;
-        }
+void Field::set_size(int x, int y) {
+    _width = x;
+    _height = y;
+    _field.resize(y);
+    for (int i = 0; i < _height; ++i) {
+        _field[i].resize(x);
     }
 }
 
-// It's supposed to be really big and sophisticated function of generation of game field, but now it's absolutely random.
+// methods /////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+// It seems to be big and sophisticated method of generation of the game field.
 void Field::generate_field() {
-    for (int i = 0; i < _field.size(); ++i) {
-        for (int j = 0; j < _field[i].size(); ++j) {
-            std::random_device device;
-            std::mt19937 gen(device());
-            int chance = gen() % 6 + 1;
-            _field[i][j] = Tile::make_tile(static_cast<TilesType>(chance));
+
+    /////////////////////////////////////////////////////////////TIME_CHECK///////////////////////////////////////////////////////////
+    std::cout << std::fixed << std::setprecision(9) << std::left;
+    auto start = std::chrono::high_resolution_clock::now();
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+    auto HOLDER = getGlobalResourceHolder<sf::Texture, std::string>;
+    for (int i = 0; i < _height; ++i) {
+        for (int j = 0; j < _width; ++j) {
+            if (j % 2)
+                _field[i][j] = Tile::make_tile(TilesType::GRASS, HOLDER().loadFromFile("../../images/grass.png", "grass"));
+            else
+                _field[i][j] = Tile::make_tile(TilesType::RIVER, HOLDER().loadFromFile("../../images/river.png", "river"));
+            _field[i][j]->get_sprite().move(sf::Vector2f(j * 32, i * 32));
         }
     }
```

Comment:
it may be better to pre-load all sprites and texture, then we will just use method 'getResource'?